Dvm

hujw77 opened the pull request.

robinwei-itering commented:
@hujw77 @AsceticBear Please create pull requests to merge to the _development_  branch.

HackFisher commented:
@hujw77 @AsceticBear @robinwei-itering 
Better to update with a checklist, and maybe we can merge a simple version for the first stage?

AsceticBear commented:
I am going to work on `crab-docs/docs/builders/tools/precompiles.md` this part this afternoon. I suggest merging the first part after STEP 1 tasks are all finished.

HackFisher commented:
@hujw77 @robinwei-itering please clean up useless images and do image replacement first to make this pull request ready for review.

Any other things before make ready for review, I moved these two to step 2.

> - [ ] `.sol` file support (Enhancement)  @JayJay1024 
> - [ ] Put the mdx code snippet into the admonitions (Enhancement) @JayJay1024

hujw77 commented:
> @hujw77 @robinwei-itering please clean up useless images and do image replacement first to make this pull request ready for review.
> 
> Any other things before make ready for review after I moved following two to step 2?
> 
> > * [ ]  `.sol` file support (Enhancement)  @JayJay1024
> > * [ ]  Put the mdx code snippet into the admonitions (Enhancement) @JayJay1024

I've replaced all the screenshots, the subscription event are not available due to a bug. It is ready for review and release a version. Please submit screenshot specifications issues in the next pr.

AsceticBear commented on docs/builders/tools/pubsub.md:
````diff
@@ -0,0 +1,145 @@
+
+```js
+const Web3 = require('web3');
+const web3 = new Web3('wss://crab-rpc.darwinia.network');
````
Change this entry point to pangolin.

AsceticBear commented on docs/builders/tools/pubsub.md:
````diff
@@ -0,0 +1,145 @@
+
+```js
+const Web3 = require('web3');
+const web3 = new Web3('wss://crab-rpc.darwinia.network');
````
And this.

hujw77 commented on docs/builders/interact/remix.md:
```diff
@@ -0,0 +1,147 @@
+
+![Verify the reduction in account balance](/images/remix/using-remix-19.png)
+
+If you encounter a failure while transferring tokens, you may need to increase the gas price and gas limit.
```
@robinwei-itering @AsceticBear Please remove this after fix the bug

robinwei-itering commented on docs/builders/interact/truffle.md:
```diff
@@ -0,0 +1,357 @@
+  },
+  compilers: {
+    solc: {
+      version: "^0.8.0"
```
Inconsistent with pragma statements @hujw77

robinwei-itering commented on docs/builders/interact/waffle-mars.md:
````diff
@@ -0,0 +1,428 @@
+```json
+{
+  "compilerType": "solcjs", // Specifies compiler to use
+  "compilerVersion": "0.8.7", // Specifies version of the compiler
````
The version assigned here differs from the text in Ln 128

robinwei-itering commented:
<img width="264" alt="Xnip2021-10-06_13-26-46" src="https://user-images.githubusercontent.com/85090505/136278545-f2f5bfde-0194-4b9d-9b33-b6a61dbd88e7.png">
This submenu is unnecessary.
